### PR TITLE
Upgrade Guzzle

### PR DESCRIPTION
**Pull request by scottsmith130**
Upgrade to Guzzle 6.3. Which means a minimum of PHP 5.5 is required now.

Also, it seems that beginning in PHP 7.2.8, unserializing a corrupt serialization will not produce an exception even if it is thrown in the custom unserialization method (as it is in the main `Sdk` class).  This means that clients will have to check that the unserialized SDK is not equal to `false` in addition to catching any unserialization exceptions that might come for other reasons (such as there being no unzip functions available for a zipped serialization).  This could be a breaking change for clients that rely only on catching exceptions while unserializing. 

So, this pull request represents a few backward-incompatible changes (minimum PHP 5.5 and new unserialization behavior).

Out of scope -  bringing anything up to PHP 5.5+ standards and syntax.

### CODE REVIEW

**Comment by scottsmith130** [comment]
Tests are not passing on PHP 7.2 yet

**Comment by scottsmith130** [comment]
Asking David to take a look since he has upgraded a clone of the PMP SDK to Guzzle 6 in the past.  And asking Ryan to take a look since he's got the most experience with this version of the SDK.

**Comment by cavis** [comment]
👍 looks good to me.  And hey, the tests caught an edge case!  IIRC, coverage is pretty good in this repo, so I'd expect any other problems to show up there as well.

**Comment by scottsmith130** [comment]
@cavis yeah, the test suite is a little unorthodox but it seems to cover well.    

I'm trying to determine whether the new version number for this should be 1.1.0 or if it should go up to 2.0.0.  I think the bump to v2 is probably warranted due to the backward-incompatible requirements/changes here, but outside of that there's not really any new or updated functionality here.

File: src/Pmp/Sdk/Http.php [hunk 1]
```diff
@@ -78,63 +83,52 @@ static public function bearerRequest($method, $url, $token = null, $data = null)
...
+            $resp = $client->request($method, $url, $opts);
         }
-        catch (\Guzzle\Http\Exception\BadResponseException $e) {
+        catch (BadResponseException $e) {
             $resp = $e->getResponse();
         }
-        catch (\Guzzle\Http\Exception\CurlException $e) {
-            // ConnectException doesn't exist yet - catch curl one manually
-            if ($e->getErrorNo() == self::CURL_COULDNT_RESOLVE_HOST) {
+        catch (GuzzleException $e) {
```
Comment (dgorsline) [inline src/Pmp/Sdk/Http.php]: You probably have reviewed my fork, and you would have seen that the fork's code catches TransferException instead.  In practice, it looks like the two are equivalent.  You may want to consider adding a trap for an empty response.  Otherwise, LGTM.

Comment (scottsmith130) [reply src/Pmp/Sdk/Http.php]: Yeah.  The trouble I had here was perhaps more with my IDE than anything else.  Something within that code path says it throws a GuzzleException, and if I didn't try to catch it here, the IDE would just simply complain.  So it seems the easiest thing to do was just catch GuzzleException here which would catch just about any exception, and we still only do something about it if it finds the curl error in the message.

Comment (scottsmith130) [reply src/Pmp/Sdk/Http.php]: It does occur to me that this will allow a GuzzleException to bubble up to the client eventually.  Before it would have been a CurlException.  So this is perhaps another change we'd need to communicate with clients.

Comment (dgorsline) [reply src/Pmp/Sdk/Http.php]: The code tells me that TransferException implements GuzzleException, and the docs and the code say that the only other exception in the Guzzle namespace that does not inherit from TransferException is SeekException.

Comment (dgorsline) [reply src/Pmp/Sdk/Http.php]: +1 for communicating to clients that the thrown exception has changed -- as of 6.0, as far as I can tell.

Comment (scottsmith130) [reply src/Pmp/Sdk/Http.php]: Ok.  I'll see about updating the README to discuss this exception being thrown (not sure that CurlException was ever communicated in the docs before).

As for the GuzzleException, it seems that all Guzzle is doing is saying some exception implementing the GuzzleException interface will be thrown, and it seems all Guzzle exceptions implement that.  So we're probably okay here.  The only reason we care specifically about this exception is because we want to catch a bad host and throw that as an exception (changing that behavior would be another breaking change).
